test/parallel/test-timers-unref-reuse-no-exposed-list.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+const timer1 = setTimeout(() => {}, 1).unref();
+assert.strictEqual(timer1._handle._list, undefined,
+                   'timer1._handle._list should be undefined');
+
+// Check that everything works even if the handle was not re-used.
+setTimeout(() => {}, 1);
+const timer2 = setTimeout(() => {}, 1).unref();
+// Note: It is unlikely that this assertion will be hit in a failure.
```

Comment:
I can't think of how this will change without removing handle re-use... Are you saying that a comment that either: might be useful or not matter at all would be better off not being there at all? Isn't that every comment ever?

That bit of code does primarily test that conditional though, that isn't a problem IMO. We have added plenty of test code for code coverage that would be similar.